server/upload.go

```diff
@@ -144,20 +144,26 @@ func (s *RESTServer) AppendFileHandler(w http.ResponseWriter, r *http.Request, p
 		f.Rollback()
 		return
 	}
+	// populate metadata fields
 	v := r.Header.Get("Content-Type")
 	if v != "" {
 		f.SetMimeType(v)
 	}
+	h := getHexadecimalHeader(r, "X-Content-SHA256")
+	if len(h) > 0 {
+		f.SetSHA256(h)
+	}
+	h = getHexadecimalHeader(r, "X-Content-MD5")
+	if len(h) > 0 {
+		f.SetMD5(h)
+	}
 }
 
 // getHexadecimalHeader returns the value for `header`, after first
 // translating it from hexadecimal to binary. If the header doesn't exist
 // or is not valid hexadecimal, returns an empty slice.
 func getHexadecimalHeader(r *http.Request, header string) []byte {
 	v := r.Header.Get(header)
-	if v == "" {
```

Comment:
Why is this being removed?

Reply:
Passing an empty string to `hex.DecodeString` returns a `nil`. So no need to test for it here. Noticed when I was adding calls to `getHexadecimalHeader`.